llvm/lib/Target/AMDGPU/SIModeRegister.cpp

```diff
@@ -430,6 +430,9 @@ void SIModeRegister::processBlockPhase3(MachineBasicBlock &MBB,
 }
 
 bool SIModeRegister::runOnMachineFunction(MachineFunction &MF) {
+  const Function &F = MF.getFunction();
+  if (F.hasFnAttribute(llvm::Attribute::StrictFP))
```

Comment:
Add a comment explaining why

Reply:
Will update

Reply:
This comment doesn't really explain why though. Should say something about how strictfp functions have user managed FP modes

Reply:
Updated, kindly review